app/services/verify_subscription_email_service.rb

```diff
@@ -1,8 +1,15 @@
 class VerifySubscriptionEmailService < ApplicationService
   class RatelimitExceededError < StandardError; end
 
-  MINUTELY_THRESHOLD = 3 # TODO: update this based on data and comment
-  HOURLY_THRESHOLD = 20 # TODO: update this based on data and comment
+  # This allows for up to 2 retries, to account for users
+  # not getting the email quickly enough and resubmitting,
+  # and for users who subscribe to several things quickly.
+  MINUTELY_THRESHOLD = 4
+
+  # This allows for up to 10 subscriptions per hour. Analysis
+  # shows 99% of subscribers have 10 or fewer active subs,
+  # so this covers someone creating all their subs in bulk.
+  HOURLY_THRESHOLD = 11
```

Comment:
Nice digging, it'd be be good to have the query used in the commit message.

Reply:
Sure, added.